cmd/google_cloud_ops_agent_engine/main.go

```diff
@@ -31,6 +32,26 @@ var (
 	stateDir = flag.String("state", "/var/lib/google-cloud-ops-agent", "path to store agent state like buffers")
 )
 
+func runStartupChecks(service string) error {
+	// Run checks in main service
+	if service == "" {
+		gceHealthChecks := healthchecks.HealthCheckRegistry{
+			healthchecks.PortsCheck{},
+			healthchecks.NetworkCheck{},
+			healthchecks.APICheck{},
+		}
+		healthCheckResults, err := gceHealthChecks.RunAllHealthChecks(*logsDir)
```

Comment:
Overall the rest of the PR LGTM. Only thing I wanted to check is here:
From what I could gather, the only error we can get is if we can't open/create the log file to write the check results to. Do we want to fail here because of that, or just write a message like "Unable to run health checks because of %v". This goes more in line with the phase 1 idea of not "short-circuiting" because of the health check.

Reply:
After discussing with @jinghan-ma and @rafaelwestphal the decision was to follow the design and phase 1 idea of not interfering with the Ops Agent execution with the Health Checks.

Removed the error reporting and it does fall back on stdout log if health check can't create the log file (which shouldn't happen since it's being created in the /var/log/google-cloud-ops-agent). Addressed in https://github.com/GoogleCloudPlatform/ops-agent/pull/1051/commits/12d6234845ecfc9cba12faab7e976bbab4276e14 .